packages/slate/src/interfaces/path.ts

```diff
@@ -391,11 +391,22 @@ export const Path = {
             }
 
             return copy.concat(p.slice(op.length))
-          } else if (
-            Path.endsBefore(onp, p) ||
-            Path.equals(onp, p) ||
-            Path.isAncestor(onp, p)
-          ) {
+          } else if (Path.endsBefore(onp, p) || Path.isAncestor(onp, p)) {
+            if (Path.endsBefore(op, p)) {
+              p[op.length - 1] -= 1
+            }
+
+            p[onp.length - 1] += 1
+          } else if (Path.equals(onp, p)) {
```

Comment:
rather than splitting out `Path.equals` with duplicate code if `op` isn't a sibling, and using an early `break`, it might be clearer to leave the code as it was and add this as a condition above?

```
// patch code
          } else if (Path.isSibling(op, p) && Path.equals(onp, p)) {
            if (Path.endsBefore(op, p)) {
              p[op.length - 1] -= 1
            } else {
              p[op.length - 1] += 1
            }
// original code
          } else if (
            Path.endsBefore(onp, p) ||
            Path.equals(onp, p) ||
            Path.isAncestor(onp, p)
          ) {
...
```

Is that the same logic? It's a bit hard to tell from the diff.